thunder/core/symbol.py

```diff
@@ -649,35 +652,8 @@ def has_tags(bsym: BoundSymbol, tags: set[OpTags]) -> bool:
 
 # Wrapper class that hashes and equates only the right hand side of a BoundSymbol for CSE.
 # That is to say, its symbol, args, and kwargs, but not its output.
-@dataclass
+@dataclass(**baseutils.default_dataclass_params)
 class BoundSymbolRHS:
-    parent: BoundSymbol
-    _frozen_kwargs: FrozenDict
-
-    def __init__(self, parent: BoundSymbol) -> None:
-        self.parent = parent
-        self._frozen_kwargs = FrozenDict(parent._var_kwargs)
-
-    @functools.cached_property
-    def _hash(self) -> int:
-        # TODO: Find a better way to identify inputs by id instead of hash.
-        if self.parent.sym.name == "unpack_trivial":
-            return id(self)
-        try:
-            return hash((self.parent.sym, self.parent._var_args, self._frozen_kwargs))
-        except:
-            return id(self)
-
-    def __hash__(self) -> int:
-        return self._hash
-
-    def __eq__(self, other: BoundSymbolRHS) -> bool:
-        if not isinstance(other, BoundSymbolRHS):
-            return False
-        if self.parent is other.parent:
-            return True
-        return (self.parent.sym, self.parent._var_args, self._frozen_kwargs) == (
-            other.parent.sym,
-            other.parent._var_args,
-            other._frozen_kwargs,
-        )
+    sym: Symbol
+    args: tuple[Any]
```

Comment:
We don't do any type checking but would this be more accurate?
```suggestion
    args: tuple[typing.Hashable]
```